src/groups/wrappers/z_bmq/z_bmqa/z_bmqa_session.h

```diff
@@ -0,0 +1,358 @@
+
+/**
+ * @brief Confirms a message with a cookie.
+ * 
+ * @param session_obj Pointer to the Session object.
+ * @param cookie Pointer to the MessageConfirmationCookie object.
+ * 
+ * @return Returns 0 upon successful confirmation.
+ */
+int z_bmqa_Session__confirmMessageWithCookie(
+    z_bmqa_Session*                         session_obj,
+    const z_bmqa_MessageConfirmationCookie* cookie);
+
+/**
+ * @brief Confirms multiple messages using a ConfirmEventBuilder.
+ * 
+ * @param session_obj Pointer to the Session object.
+ * @param builder Pointer to the ConfirmEventBuilder object containing the messages to confirm.
+ * 
+ * @return Returns 0 upon successful confirmation.
+ */
+int z_bmqa_Session__confirmMessages(z_bmqa_Session*             session_obj,
+                                    z_bmqa_ConfirmEventBuilder* builder);
+
+#if defined(__cplusplus)
+}
+
+/**
+ * @brief Custom session event handler class.
+ * 
+ * This class provides custom handling of session and message events.
+ */
+class z_bmqa_CustomSessionEventHandler
+: BloombergLP::bmqa::SessionEventHandler {
+  private:
+    z_bmqa_OnSessionEventCb onSessionEventCb; /**< Callback function for session events. */
+    z_bmqa_OnMessageEventCb onMessageEventCb; /**< Callback function for message events. */
+    void*                   data;             /**< Custom data pointer. */
+    uint64_t                mSize;            /**< Size of the custom data. */
+
+    BloombergLP::bslmt::Mutex mutex; /**< Mutex for thread safety. */
+
+  public:
+    /**
+     * @brief Constructor for z_bmqa_CustomSessionEventHandler.
+     * 
+     * @param onSessionEventCb Callback function for session events.
+     * @param onMessageEventCb Callback function for message events.
+     * @param mSize Size of the custom data.
+     */
+    z_bmqa_CustomSessionEventHandler(z_bmqa_OnSessionEventCb onSessionEventCb,
+                                     z_bmqa_OnMessageEventCb onMessageEventCb,
+                                     uint64_t                mSize);
+
+    /**
+     * @brief Destructor for z_bmqa_CustomSessionEventHandler.
+     */
+    ~z_bmqa_CustomSessionEventHandler();
+
+    /**
+     * @brief Handles session events.
+     * 
+     * @param sessionEvent Session event object.
+     */
+    void onSessionEvent(const BloombergLP::bmqa::SessionEvent& sessionEvent);
+
+    /**
+     * @brief Handles message events.
+     * 
+     * @param messageEvent Message event object.
+     */
+    void onMessageEvent(const BloombergLP::bmqa::MessageEvent& messageEvent);
+
+    /**
+     * @brief Calls a custom member function with provided arguments.
+     * 
+     * @param function Custom member function to call.
+     * @param args Arguments to pass to the custom member function.
+     */
+    void callCustomFunction(z_bmqa_SessionEventHandlerMemberFunction function,
+                            void*                                    args);
+
+    /**
```

Comment:
Why did you add this series of locking methods? The original `bmqa_session.h` did not expose any locking methods.

Reply:
Confused at this - discuss tomorrow